typed-ast dependency is not pinned

## Description

In deb-tests and staging-test-with-rebase, CI is failing with:

```
Creating Python 3 virtualenv in .venv
created virtual environment CPython3.7.13.final.0-64 in 494ms
  creator CPython3Posix(dest=/home/circleci/project/.venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/circleci/.local/share/virtualenv)
    added seed packages: pip==22.0.4, setuptools==60.10.0, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    typed-ast<1.5,>=1.4.0 from https://files.pythonhosted.org/packages/65/b3/573d2f1fecbbe8f82a8d08172e938c247f99abe1be3bef3da2efaa3810bf/typed_ast-1.4.3-cp37-cp37m-manylinux1_x86_64.whl#sha256=5feca99c17af94057417d744607b82dd0a664fd5e4ca98061480fd8b14b18d04 (from astroid==2.5.2->-r securedrop/requirements/python3/develop-requirements.txt (line 48))
make: *** [Makefile:334: ci-deb-tests] Error 1
```

I think this might be fallout from my mypy changes? unclear why it didn't get pinned...

* https://app.circleci.com/pipelines/github/freedomofpress/securedrop/4015/workflows/ad0c7371-c384-4195-b01a-599477924281/jobs/61226
* https://app.circleci.com/pipelines/github/freedomofpress/securedrop/4021/workflows/38c92e91-360e-4b02-95d8-4ef75b4a3377/jobs/61243

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -288,7 +288,7 @@ jobs:
 
   deb-tests:
     docker:
-      - image: cimg/python:3.7
+      - image: cimg/python:3.8
     environment:
       LC_ALL: C.UTF-8
       LANG: C.UTF-8

--- a/devops/gce-nested/gce-start.sh
+++ b/devops/gce-nested/gce-start.sh
@@ -25,7 +25,7 @@ function find_latest_ci_image() {
     #    --filter="family:fpf-securedrop AND name ~ ^ci-nested-virt" \
     #    --sort-by=~Name --limit=1 --format="value(Name)"
     # Return hardcoded image id to prevent newer builds from breaking CI
-    echo "ci-nested-virt-buster-1633365108"
+    echo "ci-nested-virt-bullseye-1651694527"
 }
 
 # Call out to GCE API and start a new instance, designating

--- a/securedrop/bin/dev-shell
+++ b/securedrop/bin/dev-shell
@@ -74,6 +74,7 @@ function docker_run() {
 
     # If this is a CI run, pass CodeCov's required vars into the container.
     if [ -n "${CIRCLE_BRANCH:-}" ] ; then
+        : "${CIRCLE_PULL_REQUEST:=}"
         ci_env="-e CI=true \
                 -e CIRCLECI=true \
                 -e CIRCLE_BRANCH=${CIRCLE_BRANCH:-} \

